diesel/src/sqlite/types/date_and_time/chrono.rs

```diff
@@ -375,4 +394,216 @@ mod tests {
+    fn rfc3339_datetimes_encode_correctly() {
+        let connection = connection();
+        let january_first_2000: DateTime<FixedOffset> = Utc
+            .ymd(2000, 1, 1)
+            .and_hms(0, 0, 0).into();
+
+        // select the rfc3339 string as text, extract the result as a chrono::DateTime, check no
+        // error, and then check that it deserialized to the correct datetime. All of the tests
+        // below follow this pattern.
+        let query = select(sql::<Text>("'2000-01-01T00:00:00.000000Z'"));
+        let res = query.get_result::<DateTime<FixedOffset>>(&connection);
+        assert!(res.is_ok(), "Error selecting datetime: {}", res.unwrap_err());
+
+        let dt = res.unwrap();
+        assert_eq!(dt, january_first_2000);
+
+
+        // same as above, checking that regardless of timezone storage (Z or +/-00:00), the resulting DateTime is
+        // the same.
+        let query = select(sql::<Text>("'2000-01-01T00:00:00.000000+00:00'"));
+        let res = query.get_result::<DateTime<FixedOffset>>(&connection);
+        assert!(res.is_ok(), "Error selecting datetime: {}", res.unwrap_err());
+
+        let dt = res.unwrap();
+        assert_eq!(dt, january_first_2000);
+
+
+        let query = select(sql::<Text>("'2000-01-01T00:00:00.000000-00:00'"));
+        let res = query.get_result::<DateTime<FixedOffset>>(&connection);
+        assert!(res.is_ok(), "Error selecting datetime: {}", res.unwrap_err());
+
+        let dt = res.unwrap();
+        assert_eq!(dt, january_first_2000);
+
+
+        let distant_past: DateTime<FixedOffset> = Utc
+            .ymd(0, 4, 11)
+            .and_hms(20, 00, 20).into();
+        let query = select(sql::<Text>("'0000-04-11t20:00:20.000000Z'"));
+        let res = query.get_result::<DateTime<FixedOffset>>(&connection);
+        assert!(res.is_ok(), "Error selecting datetime: {}", res.unwrap_err());
+
+        let dt = res.unwrap();
+        assert_eq!(dt, distant_past);
+
+
+        let january_first_2018 = FixedOffset::east(30*60)
+            .ymd(2018, 1, 1)
+            .and_hms_nano(12, 00, 00, 500_000);
+        let query = select(sql::<Text>("'2018-01-01t12:00:00.000500+00:30'"));
+        let res = query.get_result::<DateTime<FixedOffset>>(&connection);
+        assert!(res.is_ok(), "Error selecting datetime: {}", res.unwrap_err());
+
+        let dt = res.unwrap();
+        assert_eq!(dt, january_first_2018);
+
+
+        let distant_future = FixedOffset::west(3600)
+            .ymd(9999, 1, 8)
+            .and_hms(0, 0, 0);
+        let query = select(sql::<Text>("'9999-01-08t00:00:00.000000-01:00'"));
+        let res = query.get_result::<DateTime<FixedOffset>>(&connection);
+        assert!(res.is_ok(), "Error selecting datetime: {}", res.unwrap_err());
+
+        let dt = res.unwrap();
+        assert_eq!(dt, distant_future);
+    }
+
+    // #[test]
+    // /// Test that the sql select...where ordering functionality interops properly with rfc3339 datetimes.
```

Comment:
Those two tests should just "work" if we use `Timestamptz` as sql type. Making them work with `Text` as sql type is basically impossible.